test/common/http/codes_test.cc

```diff
@@ -229,6 +229,19 @@ TEST_F(CodeUtilityTest, PerZoneStats) {
   EXPECT_EQ(1U, cluster_scope_.counter("prefix.zone.from_az.to_az.upstream_rq_2xx").value());
 }
 
+TEST_F(CodeUtilityTest, EmptyPerZoneStats) {
+  // Empty from-zone / to-zone stats should not be emitted.
+  addResponse(200, false, false, "", "", "from_az");
+  EXPECT_EQ(0U, cluster_scope_.counter("prefix.zone.from_az..upstream_rq_completed").value());
```

Comment:
Instead of allocating the counter and checking that it's zero, can you use `TestStore::findCounterByString(...)` here?

Reply:
Sorry, a couple of things occurred to me about this test that might require more tweaks:

 * I think you should pair a negative test ("can't find stat, or stat is 0") with a positive test to be sure that stats *are* recorded where expected
 * Depending on the implementation of the string lookup method, you might actually get a counter returned for this when you don't expect it, because with your change to symbol_table_impl.cc the name will be corrected and then the stat found. So a better test might be to do the name lookup, expect that there's a counter, but get the counter's name back and test it against the expected string. I'm a little surprised that you *don't* find a match with `counter()` which will convert the string to a StatName, eliding the empty segment. So I think your test might not be spelling the stat-name correctly, even ignoring the `..`. In the new method, it will use a string-map maintained test/common/stats/stat_test_utility.cc and I'd expect the lookup to fail.

Reply:
This test doesn't really do what's intended due to this in codes.cc:
https://github.com/envoyproxy/envoy/blob/491b116c919bdeb704b536b006fe04c84fe74704/source/common/http/codes.cc#L107

So in #16675 I am just reverting this file.